### PR TITLE
Auth Docs: Handling auth yourself

### PR DESCRIPTION
**Pull request by psiinon**
Screenshot 2021-11-30 at 16-54-09 OWASP ZAP – Handling Authentication Yourself (in Automation)

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: site/content/docs/authentication/_index.md [hunk 1]
```diff
@@ -16,7 +16,8 @@ links:
     desc: you will need to understand these in order to configure authentication in ZAP
 
   - name: 'Handling authentication yourself in automation'
-    desc: Coming Soon
+    link: handling-auth-yourself
```
Comment (thc202) [inline site/content/docs/authentication/_index.md:19]: Add slash at the end?

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+The env vars are standard operating system env vars so how you will need to set them will depend on your OS and possibly your shell.
+They need to be set before you start ZAP, they cannot be set once ZAP is already running.
+
+### Scripts
+
+[Scripts](/docs/desktop/start/features/scripts/) give you complete control over all of the requests and responses that are proxied through or generated by ZAP.
+
+If you are handling authentication yourself then HTTP Sender scripts are likely to be the most useful type.
+
+The [Community Scripts](https://github.com/zaproxy/community-scripts/) Repo has a large number of
```
Comment (thc202) [inline site/content/docs/authentication/handling-auth-yourself.md:35]: Lower case repo?

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+### Scripts
+
+[Scripts](/docs/desktop/start/features/scripts/) give you complete control over all of the requests and responses that are proxied through or generated by ZAP.
+
+If you are handling authentication yourself then HTTP Sender scripts are likely to be the most useful type.
+
+The [Community Scripts](https://github.com/zaproxy/community-scripts/) Repo has a large number of 
+[HTTP Sender](https://github.com/zaproxy/community-scripts/tree/main/httpsender) scripts which you can use as the basis for your own solution.
+
+In order to use scripts in automation see the FAQ [How do you add a script to ZAP from the command line?](https://www.zaproxy.org/faq/how-do-you-add-a-script-to-zap-from-the-command-line/).
```
Comment (thc202) [inline site/content/docs/authentication/handling-auth-yourself.md:38]: Could use link without domain.

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+### Replacer rules
```
Comment (thc202) [inline site/content/docs/authentication/handling-auth-yourself.md:40]: `Rules`

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+[Replacer Rules](/docs/desktop/addons/replacer/) are an easy way to replace strings in requests and responses.
+
+The easiest way to create and test them is via the Desktop UI. You can still use them in automation - see the 
+FAQ [How do you find out what key to use to set a config value on the command line?](https://www.zaproxy.org/faq/how-do-you-find-out-what-key-to-use-to-set-a-config-value-on-the-command-line/) which includes an example for a Replacer rule.
```
Comment (thc202) [inline site/content/docs/authentication/handling-auth-yourself.md:45]: Could use link without domain.

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+If you are handling authentication yourself then HTTP Sender scripts are likely to be the most useful type.
+
+The [Community Scripts](https://github.com/zaproxy/community-scripts/) Repo has a large number of 
+[HTTP Sender](https://github.com/zaproxy/community-scripts/tree/main/httpsender) scripts which you can use as the basis for your own solution.
+
+In order to use scripts in automation see the FAQ [How do you add a script to ZAP from the command line?](https://www.zaproxy.org/faq/how-do-you-add-a-script-to-zap-from-the-command-line/).
+
+### Replacer rules
+
+[Replacer Rules](/docs/desktop/addons/replacer/) are an easy way to replace strings in requests and responses.
```
Comment (thc202) [inline site/content/docs/authentication/handling-auth-yourself.md:42]: `replace and insert`?

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
+---
+title: "Handling Authentication Yourself (in Automation)"
+tags: 
+- authentication
+- guide
+type: page
+---
+
+If you can generate an authentication token (e.g. to use in a header or cookie) and you know that your app
+will not invalidate it while you are using ZAP then one option it to handle authentication yourself.
```
Comment (kingthorin) [inline site/content/docs/authentication/handling-auth-yourself.md:10]:
```suggestion
will not invalidate it while you are using ZAP then one option is to handle authentication yourself.
```

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+### Authentication Env Vars
+
+If your app just needs one authentication token which you can generate outside of ZAP then you can use the 
+ZAP Authentication environmental variables.
+
+There are 3 of these env vars which are documented on the Desktop User Guide [Authentication](/docs/desktop/start/features/authentication/#envvars) page and reproduced here for completeness:
+
+* `ZAP_AUTH_HEADER_VALUE` - if this is defined then its value will be added as a header to all of the requests
+* `ZAP_AUTH_HEADER` - if this is defined then its value will be used as the header name - if it is not defined then the standard [Authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) header will be used
+* `ZAP_AUTH_HEADER_SITE` - if this is defined then header will only be included in sites who’s name includes its value
```
Comment (kingthorin) [inline site/content/docs/authentication/handling-auth-yourself.md:24]:
```suggestion
* `ZAP_AUTH_HEADER_SITE` - if this is defined then the header will only be included in sites whose name includes its value
```

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+[HTTP Sender](https://github.com/zaproxy/community-scripts/tree/main/httpsender) scripts which you can use as the basis for your own solution.
+
+In order to use scripts in automation see the FAQ [How do you add a script to ZAP from the command line?](/faq/how-do-you-add-a-script-to-zap-from-the-command-line/).
+
+### Replacer Rules
+
+[Replacer Rules](/docs/desktop/addons/replacer/) are an easy way to replace and insert strings in requests and responses.
+
+The easiest way to create and test them is via the Desktop UI. You can still use them in automation - see the 
+FAQ [How do you find out what key to use to set a config value on the command line?](/faq/how-do-you-find-out-what-key-to-use-to-set-a-config-value-on-the-command-line/) which includes an example for a Replacer rule.
```
Comment (kingthorin) [inline site/content/docs/authentication/handling-auth-yourself.md:45]:
```suggestion
FAQ [How do you find out what key to use to set a config value on the command line?](/faq/how-do-you-find-out-what-key-to-use-to-set-a-config-value-on-the-command-line/) which includes an example for a Replacer rule being set via the CLI.
```

File: site/content/docs/authentication/handling-auth-yourself.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+### Verification
+
+If you can define a Verification Strategies for your site then you are strongly recommended to do so.
```
Comment (kingthorin) [inline site/content/docs/authentication/handling-auth-yourself.md:49]:
```suggestion
If you can define a Verification Strategy for your site then you are strongly recommended to do so.
```

File: site/content/docs/authentication/handling-auth-yourself.md (hunk 9 above)
Comment (psiinon) [reply site/content/docs/authentication/handling-auth-yourself.md:45]: Not sure about this - the link title explicitly mentions the command line so I dont think repeating that helps, and some people might think the CLI refers to zapcli :/

**Comment by psiinon** [comment]
Fixed everything apart from the one I've queried :)

**Review by kingthorin — APPROVED** [review]

**Pull request merged**